api/deviceadm/service.go

```diff
@@ -129,38 +127,42 @@ func (s *service) UpdateDeviceStatus(ctx context.Context, uid models.UID, online
 }
 
 func (s *service) UpdatePendingStatus(ctx context.Context, uid models.UID, status, tenant, username string) error {
-	err := s.isNamespaceOnwer(ctx, tenant, username)
+	if err := s.isNamespaceOnwer(ctx, tenant, username); err != nil {
+		return ErrUnauthorized
+	}
+
+	device, err := s.store.DeviceGetByUID(ctx, uid, tenant)
 	if err != nil {
 		return err
 	}
 
-	device, _ := s.store.DeviceGetByUID(ctx, uid, tenant)
-	if device != nil {
-		if status == "accepted" {
-			sameMacDev, _ := s.store.DeviceGetByMac(ctx, device.Identity.MAC, device.TenantID, "accepted")
-
-			if sameMacDev != nil && sameMacDev.UID != device.UID {
-				if err := s.store.SessionUpdateDeviceUID(ctx, models.UID(sameMacDev.UID), models.UID(device.UID)); err != nil {
-					return err
-				}
-				if err := s.store.DeviceDelete(ctx, models.UID(sameMacDev.UID)); err != nil {
-					return err
-				}
-				if err := s.store.DeviceRename(ctx, models.UID(device.UID), sameMacDev.Name); err != nil {
-					return err
-				}
-			} else {
-				ns, err := s.store.NamespaceGet(ctx, device.TenantID)
-				if err != nil {
-					return err
-				}
-
-				if ns.MaxDevices > 0 && ns.MaxDevices <= ns.DevicesCount {
-					return ErrMaxDeviceCountReached
-				}
+	if status == "accepted" {
+		sameMacDev, err := s.store.DeviceGetByMac(ctx, device.Identity.MAC, device.TenantID, "accepted")
+		if err != nil {
+			return err
+		}
+
+		if sameMacDev.UID != device.UID {
+			if err := s.store.SessionUpdateDeviceUID(ctx, models.UID(sameMacDev.UID), models.UID(device.UID)); err != nil {
+				return err
+			}
+			if err := s.store.DeviceDelete(ctx, models.UID(sameMacDev.UID)); err != nil {
+				return err
+			}
+			if err := s.store.DeviceRename(ctx, models.UID(device.UID), sameMacDev.Name); err != nil {
+				return err
+			}
+		} else {
+			ns, err := s.store.NamespaceGet(ctx, device.TenantID)
+			if err != nil {
+				return err
+			}
+
+			if ns.MaxDevices > 0 && ns.MaxDevices <= ns.DevicesCount {
+				return ErrMaxDeviceCountReached
 			}
```

Comment:
```suggestion
		if sameMacDev.UID == device.UID {
			ns, err := s.store.NamespaceGet(ctx, device.TenantID)
			if err != nil {
				return err
			}

			if ns.MaxDevices > 0 && ns.MaxDevices <= ns.DevicesCount {
				return ErrMaxDeviceCountReached
			}

		if err := s.store.SessionUpdateDeviceUID(ctx, models.UID(sameMacDev.UID), models.UID(device.UID)); err != nil {
			return err
		}
		if err := s.store.DeviceDelete(ctx, models.UID(sameMacDev.UID)); err != nil {
			return err
		}
		if err := s.store.DeviceRename(ctx, models.UID(device.UID), sameMacDev.Name); err != nil {
			return err
		}
```

Reply:
I think a "}" is missing. It turns out that I didn't quite understand, the idea is to invert the logic of the if by the else?

Reply:
Yes, it was missing a `}`. The question was if we could drop the else and as a plus reduce one level.

Reply:
In the last version I sent, it was found that sameMacDev can be nil. Then we couldn’t drop the else.